simulation/agents.py

```diff
@@ -548,22 +559,52 @@ def make_decisions(self, model):
                 event_trigger=EventTrigger.EPC_C_UPGRADE
             )
 
-            costs = {}
+            costs_unit_and_install = {}
+            costs_fuel = {}
+            costs_subsidies = {}
+            costs_insulation = {}
+
             for heating_system in heating_system_options:
-                costs[heating_system] = self.get_total_heating_system_costs(
+
+                heating_system_costs = self.get_total_heating_system_costs(
                     heating_system, model
                 )
```

Comment:
If we changed the output to a tuple then we wouldnt need to do item assignment below which will take up a lot of time (max 5 times in this for loop, and `make_decisions` is called for each agent step).
```suggestion
                (
                    costs_unit_and_install[heating_system],
                    costs_fuel[heating_system],
                    costs_subsidies[heating_system]
                ) = self.get_total_heating_system_costs(
                    heating_system, model
                )
```